pkg/controller/controller.go

```diff
@@ -341,3 +342,17 @@ func getEmail(r *http.Request) (string, error) {
 
 	return split[1], nil
 }
+
+// GetCatagoriesDisplayData takes validators as input and returns UIData for display. If there's an error during the GetUIData RPC call, it will be thrown.
+func getCatagoriesDisplayData(validators map[string]jvspb.Validator) (map[string]*jvspb.UIData, error) {
+	displayData := make(map[string]*jvspb.UIData)
```

Comment:
Nit: you know how big this will be, save the allocation:

```suggestion
	displayData := make(map[string]*jvspb.UIData, len(validators))
```

Reply:
Updated in the latest version.